### PR TITLE
Extracted toolbar into own component

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR extracts the toolbar into an own component.

#### To Do

- [x] Readme

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Toolbar.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import Items from './Items';
+import Select from './Select';
+import toolbarStyles from './toolbar.scss';
+
+type Children = Element<typeof Controls> | Element<typeof Button> | Element<typeof Dropdown> | Element<typeof Select>;
+type Props = {
+    children: ChildrenArray<Children>,
+};
+
+@observer
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Toolbar.js]: remove

**Comment by chirimoya** [comment]
@wachterjohannes tests fail

**Comment by wachterjohannes** [comment]
@chirimoya @danrot the `lint:js` fail in files which was not changed in this PR:

* MediaCard:18
* InfiniteScroller:47

A quite strange failure https://eslint.org/docs/rules/no-trailing-spaces 

Maybe misconfiguration or update of eslint?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// @flow
+import type {ChildrenArray, Element} from 'react';
+import React from 'react';
+import Button from './Button';
+import Dropdown from './Dropdown';
+import Select from './Select';
+import toolbarStyles from './toolbar.scss';
+
+type Children = Element<typeof Button> | Element<typeof Dropdown> | Element<typeof Select>
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js]: I think we've said that we only have the `Props` type in the component itself, and move the other stuff into a separate `types.js` file.

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js [hunk 3]
```diff
@@ -20,25 +18,25 @@ const ToolbarItemTypes = {
...
-            item = (<Button {...itemConfig} />);
+            item = (<Toolbar.Button {...itemConfig} key={key} />);
     }
 
     return item;
 }
 
 @observer
-export default class Toolbar extends React.PureComponent<*> {
+export default class ToolbarContainer extends React.PureComponent<*> {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js]: At least until we haven't used suffixes for any component.

Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js]: its only for this file everywhere else the toolbar can be used with `Toolbar` see the `index.js` file in this directory.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js (hunk 2 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js]: should I create a `ItemsChildren type there?`

Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js]: Yeah, would make sense to me.

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js (hunk 3 above)
Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Toolbar.js]: Then it is even more confusing IMO. Until now the file was named after the class, and the way it is now you would see a `ToolbarContainer` component in your react dev tools, but there wouldn't be a file named like this. Could probably end in a hard time finding stuff.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js [hunk 4]
```diff
@@ -1,14 +1,11 @@
 // @flow
-import type {ChildrenArray, Element} from 'react';
+import type {ChildrenArray} from 'react';
 import React from 'react';
-import Button from './Button';
-import Dropdown from './Dropdown';
-import Select from './Select';
+import type Item from './types';
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Toolbar/Items.js]: Aren't the `{}` missing here?

**Comment by danrot** [comment]
LGTM now. Can you rebase it, to see if the build is working? The reason it is failing for you should now be fixed on develop.

**Pull request merged**